components/director/internal/domain/runtime/resolver.go

```diff
@@ -51,11 +62,23 @@ func (r *Resolver) Runtimes(ctx context.Context, filter []*graphql.LabelFilter,
 		cursor = string(*after)
 	}
 
+	tx, err := r.db.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	ctx = r.ctxvs.WithValue(ctx, appdb.DatabaseCtxKey, tx)
+
 	runtimesPage, err := r.svc.List(ctx, labelFilter, first, &cursor)
 	if err != nil {
 		return nil, err
 	}
 
+	err = tx.Commit()
```

Comment:
please check if we should call rollback explicitly, if so maybe try to implement:
```tx.RollbackUnlessCommitted()```

Reply:
Will be covered by another issue.

Reply:
not true, it is already covered :)